packages/sdk/src/oracle/services/statemachines/setActiveRequestByTransaction.ts

```diff
@@ -0,0 +1,84 @@
+import assert from "assert";
+import { Update } from "../update";
+import Store from "../../store";
+import { Log, LogDescription } from "../../types/ethers";
+import { Handlers as GenericHandlers } from "../../types/statemachine";
+import { optimisticOracle } from "../../../clients";
+
+// required exports for state machine
+export type Params = { chainId: number; transactionHash: string; eventIndex?: number };
+export type Memory = { error?: Error };
+export function initMemory(): Memory {
+  return {};
+}
+export function Handlers(store: Store): GenericHandlers<Params, Memory> {
+  const update = new Update(store);
+  return {
+    async start(params: Params, memory: Memory) {
+      const { chainId, transactionHash, eventIndex = 0 } = params;
+
+      // have to do all of this to fetch the identifier, ancData, requester and timestamp from the request
+      const provider = store.read().provider(chainId);
+      const receipt = await provider.getTransactionReceipt(transactionHash);
```

Comment:
+1 on parsing the logs directly from the transaction rather than doing a log query. This is much more efficient.